src/functions/postTypes/getPostTypeStaticProps.js

```diff
@@ -87,8 +87,12 @@ export default async function getPostTypeStaticProps(
   /* -- Handle individual posts. -- */
 
   // Handle catch-all routes.
-  const slug = Array.isArray(params.slug) ? params.slug.join("/") : params.slug;
+  let slug = Array.isArray(params.slug) ? params.slug.join("/") : params.slug;
 
+  // append stories for posts
+  if (postType === "post") {
+    slug = `stories/${slug}`;
+  }
```

Comment:
Mhh?? We're removing stories slug and then adding it? What is happening?

Reply:
@kilemensi here's whats happening. `stories/[...slug]` will match `stories/news/post-a` `stories/insights/post-c`. etc and the query objects returned are `slug: ["news", "post-a"]` `slug: ["insights", "post-c"]`. So the above method takes (& joins the slug if needed) to query word-press. But the actual uri on wordpress for a certain post is `stories/news/post-a` . So I am concatenating that here to match the full uri on wordpress.


While, the part I am slicing stories, is on the method that creates staticPaths. Again if I am on stories page folder, paths are such as { params : { slug: ["news"', "post-a"] } }. etc. So once I pull uri from wordpress, I need to slice of stories to come up with the right path.

If stories page folder was to not exist, then we won't need to do slicing and prepending

Reply:
I vote we kill the stories folder then @KhadijaMahanga.